src/js/net.js

```diff
@@ -610,11 +610,17 @@ Server.prototype.listen = function() {
   self._handle.createTCP = createTCP;
   self._handle.owner = self;
 
-  var err = self._handle.listen(backlog);
-
-  if (err) {
+  try {
+    var err = self._handle.listen(backlog);
+    if (err) {
+      self._handle.close();
+      self.emit('error', err);
+      return self;
+    }
+  } catch (err) {
     self._handle.close();
-    return self.emit('error', err);
+    self.emit('error', err);
+    return self;
   }
```

Comment:
After using `process._createUVException()`, you don't need `try...catch` to handle this error :)